indexing-service/src/main/java/io/druid/indexing/overlord/http/OverlordResource.java

```diff
@@ -483,33 +493,59 @@ public Response getCompleteTasks(@Context final HttpServletRequest req)
     final List<TaskStatus> recentlyFinishedTasks = Lists.newArrayList(
         AuthorizationUtils.filterAuthorizedResources(
             req,
-            taskStorageQueryAdapter.getRecentlyFinishedTaskStatuses(),
+            taskStorageQueryAdapter.getRecentlyFinishedTaskStatuses(maxTaskStatuses),
             raGenerator,
             authorizerMapper
         )
     );
 
-    final List<TaskResponseObject> completeTasks = Lists.transform(
-        recentlyFinishedTasks,
-        new Function<TaskStatus, TaskResponseObject>()
-        {
-          @Override
-          public TaskResponseObject apply(TaskStatus taskStatus)
-          {
-            // Would be nice to include the real created date, but the TaskStorage API doesn't yet allow it.
-            return new TaskResponseObject(
-                taskStatus.getId(),
-                DateTimes.EPOCH,
-                DateTimes.EPOCH,
-                Optional.of(taskStatus),
-                TaskLocation.unknown()
-            );
-          }
-        }
-    );
+    final List<TaskStatusPlus> completeTasks = recentlyFinishedTasks
+        .stream()
+        .map(status -> new TaskStatusPlus(
+            status.getId(),
+            taskStorageQueryAdapter.getCreatedTime(status.getId()),
+            DateTimes.EPOCH,
+            status.getStatusCode(),
+            status.getDuration(),
+            TaskLocation.unknown())
+        )
+        .collect(Collectors.toList());
+
     return Response.ok(completeTasks).build();
   }
 
+  @DELETE
+  @Path("/pendingSegments/{dataSource}")
+  @Produces(MediaType.APPLICATION_JSON)
+  public Response killPendingSegments(
+      @PathParam("dataSource") String dataSource,
+      @QueryParam("interval") String deleteIntervalString,
```

Comment:
The javadoc of ```QueryParam``` is saying

> The type T of the annotated parameter, field or property must either:
> 1. Be a primitive type
> 2. Have a constructor that accepts a single String argument
> 3. Have a static method named valueOf or fromString that accepts a single String argument (see, for example, Integer.valueOf(String))
> 4. Be List<T>, Set<T> or SortedSet<T>, where T satisfies 2 or 3 above. The resulting collection is read-only.

I guess the ```deleteInterval``` should be passed in the HTTP message body rather than ```QueryParam``` or ```PathParam``` (```PathParm``` also has similar requirements with ```QueryParam```) to make this type-safe. If so, the as-is looks better to me because it's simple.